app/uk.gov.hmrc.helptosaveproxy/health/DWPConnectionHealthCheck.scala

```diff
@@ -0,0 +1,120 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosaveproxy.health
+
+import configs.syntax._
+import akka.actor.{Actor, ActorRef, ActorSystem, PoisonPill, Props}
+import com.google.inject.Inject
+import play.api.Configuration
+import play.api.inject.ApplicationLifecycle
+import play.modules.reactivemongo.ReactiveMongoComponent
+import uk.gov.hmrc.helptosaveproxy.connectors.DWPConnector
+import uk.gov.hmrc.helptosaveproxy.health.DWPConnectionHealthCheck.DWPHealthCheckRunner
+import uk.gov.hmrc.helptosaveproxy.metrics.Metrics
+import uk.gov.hmrc.helptosaveproxy.util.lock.Lock
+import uk.gov.hmrc.helptosaveproxy.util.{Logging, PagerDutyAlerting}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.util.control.NonFatal
+
+class DWPConnectionHealthCheck @Inject() (system:            ActorSystem,
+                                          configuration:     Configuration,
+                                          metrics:           Metrics,
+                                          dWPConnector:      DWPConnector,
+                                          mongo:             ReactiveMongoComponent,
+                                          lifecycle:         ApplicationLifecycle,
+                                          pagerDutyAlerting: PagerDutyAlerting
+) extends Logging {
+
+  val name: String = "dwp-connection"
+
+  val enabled: Boolean = configuration.underlying.get[Boolean](s"health.$name.enabled").value
+
+  val lockDuration: FiniteDuration = configuration.underlying.get[FiniteDuration](s"health.$name.lock-duration").value
+
+  def newHealthCheck(): ActorRef = system.actorOf(
+    HealthCheck.props(
+      name,
+      configuration.underlying,
+      system.scheduler,
+      metrics.metrics,
+      () ⇒ pagerDutyAlerting.alert("DWP health check has failed"),
+      DWPHealthCheckRunner.props(dWPConnector, metrics),
+      DWPHealthCheckRunner.props(dWPConnector, metrics)
+    )
+  )
+
+  // make sure we only have one instance of the health check running across
+  // multiple instances of the application in the same environment
+  lazy val lockedHealthCheck: ActorRef =
+    system.actorOf(Lock.props[Option[ActorRef]](
+      mongo.mongoConnector.db,
+      s"health-check-$name",
+      lockDuration,
+      system.scheduler,
+      None,
+      _.fold(Some(newHealthCheck()))(Some(_)),
+      _.flatMap{ ref ⇒
+        ref ! PoisonPill
+        None
+      },
+      lifecycle),
+      s"health-check-$name-lock"
+    )
+
+  // start the health check only if it is enabled
+  if (enabled) {
+    logger.info(s"HealthCheck $name enabled")
+    val _ = lockedHealthCheck
+  } else {
+    logger.info(s"HealthCheck $name not enabled")
```

Comment:
prefix messages with dwp ?

Reply:
`$name` in this case expands to `dwp-connection`